### PR TITLE
(feat:#488) Add recursive proxy voting support

### PR DESCRIPTION
**Pull request by ChewingGlass**

### CODE REVIEW

File: packages/helium-admin-cli/proxy-seasons.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[
```
[the remark this answers is not in the document]
Comment (ChewingGlass) [reply packages/helium-admin-cli/proxy-seasons.json:1]: We've got to figure this out for a number of configs. I think eventually we'll need to create an endpoint to update a subsection of the array so we can push more than 1kb of information. Similar to the way candymachine used to work when it was getting loaded up. For now I'm not as worried about it, as I suspect protocol is going to change enough in the next 5 years that it no longer matters.

File: programs/voter-stake-registry/src/state/registrar.rs [hunk 2]
```diff
@@ -21,7 +21,8 @@ pub struct Registrar {
 
   // Empty bytes for future upgrades.
   pub reserved1: [u8; 4],
-  pub reserved2: [u64; 7], // split because `Default` does not support [u8; 60]
+  pub reserved2: [u64; 3], // split because `Default` does not support [u8; 60]
```
[the remark this answers is not in the document]
Comment (ChewingGlass) [reply programs/voter-stake-registry/src/state/registrar.rs:24]: It shouldn't, as it's moving by the same bytes as we're adding below. Def worth double checking my math though.

File: packages/account-fetch-cache-hooks/src/hooks/useAccounts.ts [hunk 3]
```diff
@@ -125,65 +104,84 @@ export function useAccounts<T>(
...
+    }
+
+    return [];
+  }, [eagerResult, result]);
+
+  useEffect(() => {
+    if (error) {
+      console.error(error);
+    }
+  }, [error]);
```
Comment (bryzettler) [inline packages/account-fetch-cache-hooks/src/hooks/useAccounts.ts:179]: we want to do anything besides `console.error`ing this?

File: packages/account-fetch-cache-hooks/src/hooks/useAccounts.ts [hunk 4]
```diff
@@ -195,23 +193,43 @@ export function useAccounts<T>(
...
+      publicKey: PublicKey;
+    }[]
+  >(accounts || []);
 
   useEffect(() => {
-    const keySet = new Set(keys ? keys.map((k) => k.toBase58()) : []);
+    if (watchedAccounts != accounts) {
+      setWatchedAccounts(accounts);
+    }
+  }, [accounts]);
```
Comment (bryzettler) [inline packages/account-fetch-cache-hooks/src/hooks/useAccounts.ts:210]: This check safe to have in a `useEffect`? wont it always return true and fire?

**Review by bryzettler — APPROVED** [review]

**Pull request merged**